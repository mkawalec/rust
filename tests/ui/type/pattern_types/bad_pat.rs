#![feature(pattern_types)]
#![allow(incomplete_features)]

type NonNullU32_2 = u32 is 1..=;
//~^ ERROR: inclusive range with no end
type Positive2 = i32 is 0..=;
//~^ ERROR: inclusive range with no end
type Wild = () is _;
//~^ ERROR: wildcard patterns are not permitted for pattern types

fn main() {}
