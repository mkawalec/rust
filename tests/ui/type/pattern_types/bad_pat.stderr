error[E0586]: inclusive range with no end
  --> $DIR/bad_pat.rs:4:29
   |
LL | type NonNullU32_2 = u32 is 1..=;
   |                             ^^^ help: use `..` instead
   |
   = note: inclusive ranges must be bounded at the end (`..=b` or `a..=b`)

error[E0586]: inclusive range with no end
  --> $DIR/bad_pat.rs:6:26
   |
LL | type Positive2 = i32 is 0..=;
   |                          ^^^ help: use `..` instead
   |
   = note: inclusive ranges must be bounded at the end (`..=b` or `a..=b`)

error: "wildcard patterns are not permitted for pattern types"
  --> $DIR/bad_pat.rs:8:19
   |
LL | type Wild = () is _;
   |                   ^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0586`.
